File: examples/python/example1.py
import sys
sys.path.append("/home/yuri/global-ruby/sources/python")

from global_python import Global

def collect_info():
    with open("/proc/cpuinfo") as f:
        return sum(1 for line in f if "processor" in line)

hosts = ["127.0.0.1"]
#hosts = ["130.162.50.40", "127.0.0.1"]

total_cores = sum(Global.run_now(collect_info, host) for host in hosts)
print(f"Total cores on all hosts: {total_cores}")

File: examples/ruby/example1.rb

require './lib/global-ruby'

@prefix = "LOG"

def header2
  @prefix
end

def header
  puts header2 + ": " + "scanning #{`hostname`.strip} with variable #{@data}..."
end

def msg(text)
  puts "LOG: #{text}"
end

def collect_info
  header
  `cat /proc/cpuinfo  | grep processor | wc -l`.to_i
end

@data = 3

hosts = [ '127.0.0.1', '130.162.50.40' ] # Specify any SSH-accessible hosts
puts "Total cores: #{hosts.sum { |host| Global.run(binding, host, :collect_info) }}"

#hosts.each { |h| Global.run(binding, h, :collect_info) }
#hosts.each { |host| puts Global.run(binding, host, File.method(:exist?), "/tmp") }

